integration-tests/src/test/java/oracle/kubernetes/operator/ItModelInImageConfigUpdate.java

```diff
@@ -153,6 +153,62 @@ public void testMiiConfigUpdateNonExistJdbc() throws Exception {
     LoggerHelper.getLocal().log(Level.INFO, "SUCCESS - " + testMethodName);
   }
 
+  /**
+   * Create a domain without a JDBC DS using model in image and having configmap
+   * in the domain.yaml before deploying the domain. After deploying the domain crd,
+   * create a new image with diff tag name and model files that define a JDBC DataSource
+   * and update the domain crd to change image name to reload the model,
+   * generate new config and initiate a rolling restart.
+   *
+   * @throws Exception if domain creation, config update or test veriofication fails
+   */
+  @Test
+  public void testMiiConfigUpdateNonJdbcImage() throws Exception {
+    Assumptions.assumeTrue(QUICKTEST);
+    String testMethodName = new Object() {
+    }.getClass().getEnclosingMethod().getName();
+    logTestBegin(testMethodName);
+    LoggerHelper.getLocal().log(Level.INFO,
+        "Creating Domain & waiting for the script to complete execution");
+    boolean testCompletedSuccessfully = false;
+    try {
+      // create Domain w/o JDBC DS using the image created by MII
+      boolean createDS = false;
+      createDomainUsingMii(createDS);
+
+      // create image with a new tag to update config
+      Map<String, Object> domainMap = domain.getDomainMap();
+      final String imageName = (String) domainMap.get("image") + "_nonjdbc";
+      final String wdtModelFile = "model.jdbc.yaml";
+      final String wdtModelPropFile = "model.jdbc.properties";
+      createDomainImage(domainMap, imageName, wdtModelFile, wdtModelPropFile);
+
+      // update domain yaml with new image tag and
+      // apply the domain yaml, verify domain rolling-restarted
+      modifyDomainYamlWithImageName(domain, domainNS, imageName);
+
+      // verify the test result by checking updated config file on server pod
+      verifyJdbcUpdate();
+
+      // verify that JDBC DS is created by checking JDBC DS name and read timeout
+      LoggerHelper.getLocal().log(Level.INFO, "Verify that JDBC DS is created");
+      Set<String> jdbcResourcesToVerify = new HashSet<String>();
+      jdbcResourcesToVerify.add("datasource.name.1=" + dsName);
+      jdbcResourcesToVerify.add("datasource.readTimeout.1=" + readTimeout_2);
+
+      final String destDir = getResultDir() + "/samples/model-in-image-update";
+      verifyJdbcResources(jdbcResourcesToVerify, destDir);
+
+      testCompletedSuccessfully = true;
```

Comment:
is it just me, or is there not a single assertion in this whole test?

Reply:
assertion is inside of verifyJdbcResources(jdbcResourcesToVerify, destDir);

  private void verifyJdbcResources(Set<String> jdbcResourcesSet, String destDir)  {
    // verify JDBC DS props via WLST on server pod
    String jdbcResources = getJdbcResources(destDir);

    Iterator<String> iterator = jdbcResourcesSet.iterator();
    while (iterator.hasNext()) {
      String prop = iterator.next();
      Assertions.assertTrue(jdbcResources.contains(prop), prop + " is not found");
      LoggerHelper.getLocal().log(Level.INFO, prop + " exists");
    }
  }

Reply:
it feels like there should be more than that, i think we should be checking each operation as we go, to make sure it is behaving as we expect.  your javadoc reads like there are several steps - we should check as we go to make sure they all work, please